src/lib/components/Swap/Status/StatusDialog.tsx

```diff
@@ -23,17 +25,22 @@ const TransactionRow = styled(Row)`
   flex-direction: row-reverse;
 `
 
+const Link = styled(EtherscanLink)`
+  text-decoration: none;
+`
+
 function ElapsedTime({ tx }: { tx: Transaction<SwapTransactionInfo> }) {
   const [elapsedMs, setElapsedMs] = useState(0)
   useInterval(
     () => {
-      if (tx.info.response.timestamp) {
-        setElapsedMs(tx.info.response.timestamp - tx.addedTime)
+      if (tx.receipt && tx.info.response.timestamp) {
+        setElapsedMs(tx.addedTime - tx.info.response.timestamp)
```

Comment:
wouldn't this be a negative number?

Reply:
oh, tricky bug. Wasn't showing up because the null below halts the timer at the last `Date.now() - tx.addedTime` and it *looks* correct because this block doesn't run. Good catch.